src/index.ts

```diff
@@ -160,10 +170,26 @@ export class StatsStore {
     if (this.optOut || this.isDevMode) {
       return;
     }
-    const stats = await this.getDailyStats();
+
+    // If multiple instances of `telemetry` are being run from different
```

Comment:
When we discussed this synchronously, I believe you mentioned `localStorage` would itself implement a mutex around accesses to any of its keys, and that the lock would be released before the next tick of the loop. Are we taking advantage of those semantics here? If so, can we document that as well (maybe linking to some resource describing that behavior)?

Reply:
Yes exactly, I've just double-checked more deeply an this seems to be quite complex thing 😅

In [the W3C Storage standard](https://www.w3.org/TR/webstorage/#threads) there's some information about it.

But then there's [this article](https://balpha.de/2012/03/javascript-concurrency-and-locking-the-html5-localstorage/) (quite old imho), which states that `localStorage` is not thread safe in most browsers.

After researching a bit, seems like at the time (2012) most of the engines didn't implement the mutex (to be fair, most of the browsers didn't run tabs in separate threads), so this could be an issue back then. It even seems that Chrome removed temporarily the mutexes for performance reasons. Today, with the predominance of chromium and with the fact that Atom runs on `electron` this doesn't seem to be a problem.

Nowadays, `chromium` (and therefore `electron`) seems to have [a few mechanism to avoid the race conditions](https://github.com/chromium/chromium/blob/33fe7900769db1a4a005de6004253f75a28dd614/third_party/blink/renderer/modules/storage/storage_area_map.cc) but honestly I haven't gone deep into their code to fully understand what's doing.

To be completely sure, I've just tried by brute force to cause a race condition both on Atom and Chrome and was unable to do so. This is the code that I've used:

```js
// window 1
const timer = setInterval(() => {
  if (localStorage.getItem('foo') === '1') {
    return;
  }  
  console.log('trying to cause a race condition');

  let i = 4000000;
  while (i--) {
    if (localStorage.getItem('foo') === '1') {
      console.error('Race condition!');
      clearInterval(timer);
      break;
    }
  }
  console.log('no race condition yet...');
}, 1000);

// =========================
// window 2
let value = '1';
setInterval(() => {
  value = value === '1' ? '0' : '1';
  localStorage.setItem('foo', value);
}, 1);
```

I've run this super abusive code for almost an hour with no race condition happening, so I'm pretty sure that even if there was a possibility of a race condition it won't ever happen in the metrics scenario.

I'm gonna add a comment linking to the standard though 😄